cli/core/utils.go

```diff
@@ -161,6 +163,68 @@ func GetCurrentCheckpoint(eigenpodAddress string, client *ethclient.Client) (uin
 	return timestamp, nil
 }
 
+// Fetch and return the current checkpoint timestamp for the pod
+// If the checkpoint exists (timestamp != 0), also return the beacon state for the checkpoint
+// If the checkpoint does not exist (timestamp == 0), return the head beacon state (i.e. the state we would use "if we start a checkpoint now")
+func GetCheckpointTimestampAndBeaconState(
```

Comment:
reminder for later - prob "ProvableBeaconStateForCheckpoint" or something like that